src/project.cpp

```diff
@@ -1571,13 +1558,8 @@ void Project::loadTilesetMetatiles(Tileset* tileset) {
             Metatile *metatile = new Metatile;
             int index = i * (2 * 4 * num_layers);
             for (int j = 0; j < 4 * num_layers; j++) {
-                uint16_t word = data[index++] & 0xff;
-                word += (data[index++] & 0xff) << 8;
-                Tile tile;
-                tile.tileId = word & 0x3ff;
-                tile.xflip = (word >> 10) & 1;
-                tile.yflip = (word >> 11) & 1;
-                tile.palette = (word >> 12) & 0xf;
+                Tile tile(static_cast<unsigned char>(data[index++])
```

Comment:
I'm worried these two `++` operator usages are actually undefined behavior, regarding order.  I don't know the answer, but it makes me squeamish.

Reply:
Good call, my understanding is that unlike logical or there's no guaranteed order here.